src/main/java/org/killbill/billing/plugin/adyen/client/AdyenConfigProperties.java

```diff
@@ -238,10 +249,10 @@ public String getMerchantAccount(final String countryIsoCode) {
         if (countryToMerchantAccountMap.isEmpty()) {
             return merchantAccounts;
         } else if (countryIsoCode == null) {
-            // In case no country is specified, but the user configured the merchant accounts per country, take the first one
-            return countryToMerchantAccountMap.values().iterator().next();
+            // In case no country is specified, but the user configured the merchant accounts per country, take the fallback one or the first one.
+            return MoreObjects.firstNonNull(fallBackMerchantAccount, countryToMerchantAccountMap.values().iterator().next());
```

Comment:
An NPE will now be thrown if that country isn't configured and I'm not sure the stacktrace will be that useful.

Could you handle that case specifically instead? Maybe throw an `IllegalStateException` with a message indicating the country?

Reply:
Good catch. Done in 36984ba